File: #0287.find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {

        int result = 0;
        for (auto n : nums) {
            if (nums[abs(n) - 1] < 0) {
                result = std::abs(n);
                break;
            }
            nums[std::abs(n) - 1] =- nums[std::abs(n) - 1];
        }
        return result;
    }
};

File: #0287.find-the-duplicate-number.rs
impl Solution {
    pub fn find_duplicate(nums: Vec<i32>) -> i32 {
        let mut seen = [false; 1e5 as usize];
        for num in nums {
            let idx = num as usize - 1;
            if seen[idx] {
                return num;
            }
            seen[idx] = true;
        }
        0
    }
}
